src/features/kernel.md

```diff
@@ -8,3 +8,131 @@ The CheriBSD kernel can be compiled either as hybrid or pure-capability code:
   memory protection internally in the kernel, protecting against memory-safety
   vulnerabilities in components such as the network stack and system-call
   layer.
+
+A number of precompiled kernels are provided in the CheriBSD release, enabling
+or disabling debugging and revocation in combination with hybrid or
+pure-capability compilations.
+In general, we recommend that performance benchmarking be done only with
+`NODEBUG` kernels; see [Benchmarking guidance](../benchmarking/README.md) for
+further details.
+
+## Precompiled kernels in CheriBSD 23.11
+
+CheriBSD ships with a number of precompiled kernels that enable and disable
+various relevant kernel configuration options.
+Kernel configuration files themselves may be found in `sys/amd64/conf` in the
+CheriBSD source tree, and will have names such as `GENERIC-MORELLO`.
+The following kernel features may be selected from amongst the precompiled
+kernels:
+
+| Option name   | Description |
+|---------------|-------------|
+| `PURECAP`     | Compiled with internal memory safety |
+| `NODEBUG`     | Compiled without kernel debugging features |
+| `NOCAPREVOKE` | Compiled without userspace heap temporal safety |
+
+The default kernel uses the `GENERIC-MORELLO` kernel
+configuration, and is a hybrid kernel with debugging and userspace heap
+temporal safety enabled.
+It is unsuitable for benchmarking; instead, use a `NODEBUG` kernel
+configuration.
+The default kernel is expected to change to a pure-capability kernel in future
+CheriBSD releases.
+
+Kernel directory names are in the form `/boot/kernel*`; the default kernel is
+named `/boot/kernel`.
+
+```
+# cd /boot
+# ls -d kernel*
+kernel
+kernel.GENERIC-MORELLO-NOCAPREVOKE
+kernel.GENERIC-MORELLO-NOCAPREVOKE-NODEBUG
+kernel.GENERIC-MORELLO-NODEBUG
+kernel.GENERIC-MORELLO-PURECAP
+kernel.GENERIC-MORELLO-PURECAP-NOCAPREVOKE
+kernel.GENERIC-MORELLO-PURECAP-NOCAPREVOKE-NODEBUG
+kernel.GENERIC-MORELLO-PURECAP-NODEBUG
+```
+
+## Identifying the currently booted kernel
+
+The `uname(1)` command prints information on the kernel configuration.
+The following command-line options may be of use:
+
+| Option name | Description                         |
+|-------------|-------------------------------------|
+| `-a`        | Print numerous details              |
+| `-i`        | Print the kernel configuration name |
+
+For example:
+
+```
+# uname -i
+GENERIC-MORELLO
+```
+
```

Comment:
```suggestion

Additionally, the current kernel directory path is stored in the `kern.bootfile` sysctl variable
that you can read with the `sysctl(8)` utility. For example:

# sysctl kern.bootfile
kern.bootfile: /boot/kernel.GENERIC-MORELLO

```

Unfortunately, GitHub does not correctly handle suggestions with code snippets so you'd have to add a code block after accepting it...